Providers/Profile.swift

```diff
@@ -782,8 +806,59 @@ public class BrowserProfile: Profile {
             }
         }
 
+        @objc func onStartSyncing(notification: NSNotification) {
+            syncDisplayState = .InProgress
+        }
+
         @objc func onFinishSyncing(notification: NSNotification) {
-            self.lastSyncFinishTime = NSDate.now()
+            if let syncState = syncDisplayState where syncState == .Good {
+                self.lastSyncFinishTime = NSDate.now()
+            }
+        }
+
+        private func displayStateForSyncResult(result: SyncResult) -> SyncDisplayState {
+            guard result.value.isSuccess,
+            let syncState = result.value.successValue else {
+                return SyncDisplayState.Bad(message: Strings.FirefoxSyncFailedTitle)
+            }
+
+            return displayStateForSyncState(syncState)
+        }
+
+        private func displayStateForEngineResults(result: Maybe<EngineResults>?) -> SyncDisplayState {
+            guard let result = result else {
+                return .Good
+            }
+            guard result.isSuccess,
```

Comment:
Nit: Don't need the `isSuccess` check if you're conditionally unwrapping `successValue`